ccgains/bags.py

```diff
@@ -636,15 +643,53 @@ def deposit(self, dtime, currency, amount, fee, exchange):
             log.info("Taxable loss due to fees: %.3f %s",
                      prof, self.currency)
 
+    def pick_bag(self, exchange, currency):
+        """Pick bag from bag queue according to mode
+
+        :param exchange: (string) The unique name of the
+            exchange/wallet where the funds that are being spent are
+            taken from.
+        :param currency: (string) The currency being spent.
+        """
+        def pick_first():
+            i = 0
```

Comment:
Hmm, I wonder if we should optimize this, or is it premature? ;)
Don't start at 0 every time this function is called, but rather at optional argument `start_index=0`.
Then, when a large amount of first bags encountered don't have the proper currency, they don't have to be iterated over each and every time this function is called.

Reply:
OK, `start_index=0` would break 'LIFO' mode. So rather parameter `start_index=None` and then
```
if start_index is None:
    start_index = len(self.bags[exchange]) - 1 if self.mode == "LIFO" else 0
```

Reply:
Probably it's better to use -1 as staring index for LIFO, so index caching will work too.